railties/test/application/rake/dbs_test.rb

```diff
@@ -300,12 +308,37 @@ def db_test_load_structure
           Dir.chdir(app_path) do
             database_path = `bin/rails db:setup`
             assert_equal "development.sqlite3", File.basename(database_path.strip)
+            p File.basename(database_path.strip)
           end
         ensure
           ENV["RAILS_ENV"] = @old_rails_env
           ENV["RACK_ENV"] = @old_rack_env
         end
       end
+
+      test "db:load_schema_or_structure loads structure if schema_format is sql" do
+        add_to_config "config.active_record.schema_format = :sql"
+        require "#{app_path}/config/environment"
+        Dir.chdir(app_path) do
+          db_structure_dump
+          `bin/rails environment db:drop`
+          `bin/rails db:load_schema_or_structure`
+          require "#{app_path}/app/models/book"
+          assert_equal 0, Book.count
+        end
+      end
+
+      test "db:load_schema_or_structure loads schema if schema_format is ruby" do
+        add_to_config "config.active_record.schema_format = :ruby"
+        require "#{app_path}/config/environment"
+        Dir.chdir(app_path) do
+          db_schema_dump
+          `bin/rails environment db:drop`
+          `bin/rails db:schema:load`
```

Comment:
It looks like the `db:load_schema_or_structure` task should be run here instead.